VIDEO_2_SCRIPT.md

```diff
@@ -0,0 +1,157 @@
+
+- **Simple**: design is straightforward, optimizing organizational capacity.
+
+The disadvantages of this architecture model are that without communication standards between applications, a lot of coding time is required. Applications with a high level of data transfer or with a short life span are not compatible with this model. At the level of protocols and services, it is arguably a more expensive system than others we have seen previously.
+
+## Event-driven architecture
+
+Event-driven architecture is a software model and architecture used to design applications. In such a system, the capture, communication, processing, and permanence of events are the core structure of the solution. This differs from the traditional request-based model.
+
+But, what’s an event? Let’s go backwards…
+
+Events are those significant changes in the state of a system's hardware or software. An event and its notification are not the same things: the latter is a message that the system sends to communicate to another part of the system that a certain event occurred.
+
+Events can originate from internal or external stimuli. They can be generated by a user's activity, for example, when he clicks the mouse or presses a key; from an external source, as in the case of a sensor; or from the system, for instance, when a program is loaded.
+
+This type of architecture is composed of event producers and event consumers. The former detects events and represents them as messages. It does so without knowing the event consumer or the result that the latter will generate. 
+
+Once an event is detected, it is transmitted from the producer to the consumers through event channels. There it is processed asynchronously thanks to a platform for this purpose. When an event occurs, it must be reported to the consumers, who could process it or simply receive it. 
+
+The processing platform will execute the appropriate response for the event and send the activity to the corresponding consumers. This downstream activity corresponds to where the result of the event will be seen. 
+
+Benefits of event-driven architecture:
+
+- **Scalability**: Scalability is one of the strongest points of this architecture since it allows each consumer to scale independently, thus reducing the coupling between components.
+
+- **Deployment**: Due to the low coupling between components, deployment is possible without worrying about dependencies or preconditions.
+
+- **Flexibility**: EDA allows rapid response to a changing environment, because each event processing component has a single responsibility and is completely decoupled from the others.
+
+The most significant disadvantages associated with this model are the difficult testability of the tests, being an asynchronous model adds more complexity to this part. The lack of guarantee of obtaining a response to the sender and the scarcity of recovery support in case of partial failure are the most significant drawbacks.
```

Comment:
```suggestion
The most significant disadvantages associated with this model are the difficult testability of the tests. Being an asynchronous model adds more complexity to this part. The lack of guarantee of obtaining a response to the sender and the scarcity of recovery support in case of partial failure is the most significant drawback.
```